ZBUG-3364:Added test case to validate service.PERM_DENIED is not coming when sending email as distribution list and using persona.

pallavikhairnar21 opened the pull request:
ZBUG-3364:Added test case to validate error-service.PERM_DENIED is not coming when sending email as distribution list and using persona.
Test case 1: Verify sendAsDistList DL rights and no service.PERM_DENIED error when sending email as distribution list and using persona
Steps:
                        1. Login into Admin.
			2. Grant Permission "sendAsDistList" to Account1 on Distribution list.
			3. Login into Account1.
			4. Send DiscoverRightsRequest to discover grants granted.
			5. Send email with From: set to DL and Verify message details using GetMsgRequest
			6. Create a persona for DL 
			7. Send an email using persona and verify service.PERM_DENIED error is not encountered
			8. Verify message details using GetMsgRequest
Test case 2: Verify sendOnBehalfOfDistList DL rights and no service.PERM_DENIED error when sending email as distribution list and using persona
Steps:
                        1. Login into Admin.
			2. Grant Permission "sendOnBehalfOfDistList" to Account1 on Distribution list.
			3. Login into Account1.
			4. Send DiscoverRightsRequest to discover grants granted.
			5. Send email with From: set to DL and Verify message details using GetMsgRequest
			6. Create a persona for DL 
			7. Send an email using persona and verify service.PERM_DENIED error is not encountered
			8. Verify message details using GetMsgRequest
I will add additional test cases in next automation ticket..

Test automation result:
XML: C:\soap\zm-soap-harness\data\soapvalidator\Mail\Persona\SendMailUsingDistributionListPersona.xml


-----------------------------REQUEST PERFORMANCE SUMMARY---------------------------------

Request Name                              ExecutionTime(msec)  ExecutionCount  AvgExecutionTime(msec)

AuthRequest                                       10237          7           1462
CreateAccountRequest                              14214          3           4738
DiscoverRightsRequest                              3172          2           1586
SearchRequest                                      3272          2           1636
PingRequest                                        1941          1           1941
GrantRightRequest                                  3069          2           1534
CreateDistributionListRequest                      3155          2           1577
AddDistributionListMemberRequest                   2907          2           1453
CreateIdentityRequest                              2964          2           1482
GetMsgRequest                                      9037          6           1506
SendMsgRequest                                     5533          4           1383


-------------------SUMMARY---------------------

PASS 0001   1/1  1941ms PingRequest                 	Ping
PASS 0002   1/1  1423ms AuthRequest                 	acct1_setup
PASS 0003   2/2 10330ms CreateAccountRequest        	acct1_setup
PASS 0004   2/2  1980ms CreateAccountRequest        	acct1_setup
PASS 0005   2/2  1904ms CreateAccountRequest        	acct1_setup
PASS 0006   1/1  1623ms CreateDistributionListRequest	acct1_setup
PASS 0007   1/1  1244ms AddDistributionListMemberRequest	acct1_setup
PASS 0008   1/1  1532ms CreateDistributionListRequest	acct1_setup
PASS 0009   1/1  1663ms AddDistributionListMemberRequest	acct1_setup
PASS 0010   1/1  1242ms AuthRequest                 	sendMailAsDL
PASS 0011   1/1  1433ms GrantRightRequest           	sendMailAsDL
PASS 0012   2/2  1333ms AuthRequest                 	sendMailAsDL
PASS 0013   2/2  1533ms DiscoverRightsRequest       	sendMailAsDL
PASS 0014   2/2  1436ms SendMsgRequest              	sendMailAsDL
PASS 0015   4/4  1539ms GetMsgRequest               	sendMailAsDL
PASS 0016   3/3  1325ms CreateIdentityRequest       	sendMailAsDL
PASS 0017   2/2  1427ms SendMsgRequest              	sendMailAsDL
PASS 0018   5/5  1440ms GetMsgRequest               	sendMailAsDL
PASS 0019   2/2  1637ms AuthRequest                 	sendMailAsDL
PASS 0020   2/2  1636ms SearchRequest               	sendMailAsDL
PASS 0021   5/5  1635ms GetMsgRequest               	sendMailAsDL
PASS 0022   1/1  1636ms AuthRequest                 	sendMailOnBehalfOfDL
PASS 0023   1/1  1636ms GrantRightRequest           	sendMailOnBehalfOfDL
PASS 0024   2/2  1330ms AuthRequest                 	sendMailOnBehalfOfDL
PASS 0025   2/2  1639ms DiscoverRightsRequest       	sendMailOnBehalfOfDL
PASS 0026   2/2  1325ms SendMsgRequest              	sendMailOnBehalfOfDL
PASS 0027   4/4  1448ms GetMsgRequest               	sendMailOnBehalfOfDL
PASS 0028   3/3  1639ms CreateIdentityRequest       	sendMailOnBehalfOfDL
PASS 0029   2/2  1345ms SendMsgRequest              	sendMailOnBehalfOfDL
PASS 0030   5/5  1646ms GetMsgRequest               	sendMailOnBehalfOfDL
PASS 0031   2/2  1636ms AuthRequest                 	sendMailOnBehalfOfDL
PASS 0032   2/2  1636ms SearchRequest               	sendMailOnBehalfOfDL
PASS 0033   5/5  1329ms GetMsgRequest               	sendMailOnBehalfOfDL


script_parsable:	2	0



TEST CASE Results: 2(PASS) - 0(Fail)


REQUEST/RESPONSE Results: 33(PASS) - 0(Fail)

deepakzimb reviewed (approved):
LGTM

jiteshsojitra commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+		</t:response>
+	</t:test>
+	
+	<!-- Verify message details using GetMsgRequest -->
```
indentation

pallavikhairnar21 replied:
Corrected indentation

jiteshsojitra commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+	</t:test>
+
+	
+	<!-- Verify message details using GetMsgRequest -->
+    <t:test id="GetMsgRequest">
+        <t:request>
+            <GetMsgRequest xmlns="urn:zimbraMail">
+                <m id="${message.id2}"/>
+            </GetMsgRequest>
+        </t:request>
+       <t:response>
```
indentation

pallavikhairnar21 replied:
Corrected indentation

jiteshsojitra replied:
entire file has indentation problem, please use tabs everywhere if that is followed.

jiteshsojitra commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+<t:tests xmlns:t="urn:zimbraTestHarness">
+
+<t:property name="test_account1.name" value="test1.${TIME}.${COUNTER}@${defaultdomain.name}"/>
+<t:property name="test_account2.name" value="test2.${TIME}.${COUNTER}@${defaultdomain.name}"/>
```
is this account used for delegated testcases?

pallavikhairnar21 replied:
test_account1.name is the account which is part of DL and has sendAsDistList right. I will add few more test cases in the next sprint so keep test_account2.name as well ZCS-13336. If needed I can delete it - test_account2.name.

jiteshsojitra commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+<t:tests xmlns:t="urn:zimbraTestHarness">
+
+<t:property name="test_account1.name" value="test1.${TIME}.${COUNTER}@${defaultdomain.name}"/>
+<t:property name="test_account2.name" value="test2.${TIME}.${COUNTER}@${defaultdomain.name}"/>
+<t:property name="test_account3.name" value="test3.${TIME}.${COUNTER}@${defaultdomain.name}"/>
```
is this account used for delegated testcases?

jiteshsojitra replied:
is this account used for delegated testcases?

jiteshsojitra commented:
Bugs/ZBUG-3364.xml

Please stop using adding testcase under Bugs, create proper named file under valid location for e.g. SendMailUsingPersonaDelegation or SendMailUsingPersonaDistributionList something like this the way SOAP harness follows.

jiteshsojitra commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+			</SendMsgRequest>
+		</t:request>
+		<t:response>
+			<t:select path="//zimbra:Code" match="service.PERM_DENIED" emptyset="1"/>
+			<t:select path="//mail:SendMsgResponse/mail:m" attr="id" set="message.id1"/>
```
Please verify from headers (who is sending mail / send as).

pallavikhairnar21 commented:
> Bugs/ZBUG-3364.xml
> 
> Please stop using adding testcase under Bugs, create proper named file under valid location for e.g. SendMailUsingPersonaDelegation or SendMailUsingPersonaDistributionList something like this the way SOAP harness follows.

Added Persona folder in Mail and added SendMailUsingDistributionListPersona in it..

prashantajari commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+<t:tests xmlns:t="urn:zimbraTestHarness">
+
+<t:property name="test_account1.name" value="test1.${TIME}.${COUNTER}@${defaultdomain.name}"/>
+<t:property name="test_account2.name" value="test2.${TIME}.${COUNTER}@${defaultdomain.name}"/>
```
Lets not create too many test accounts. I think, we can restrict account to account1, account2 (as DL members) and account3 as recipient user.

prashantajari commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+            </t:request>
+            <t:response>
+                <t:select path="//admin:CreateAccountResponse/admin:account" attr="id"  set="test_account1.id"/>
+                <t:select path='//admin:CreateAccountResponse/admin:account/admin:a[@n="zimbraMailHost"]' set="test_acct1.server"/>
```
test_acct1.server and similar values are not used in test and should be removed

prashantajari commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+			</t:select>
+			</t:response>
+	</t:test>
+</t:test_case>
```
Please add validation at recipient end to check from/to address and other message details

prashantajari commented on data/soapvalidator/Mail/Bugs/ZBUG-3364.xml:
```diff
@@ -0,0 +1,516 @@
+			</t:select>
+			</t:response>
+    </t:test>
+	
```
Please add validation at recipient end to check from/to address and other message details

prashantajari reviewed (changes requested):
Minor changes requested

pallavikhairnar21 commented:
> Minor changes requested

Fixed comments provided by Prashant and Jitest, please check review again

jiteshsojitra commented:
1. Still there were many indentation issues which was not fixed.
2. There should be only 1 commit with rebased with master/develop.
